更新最新代码后，models路径不支持子目录

今天合并了官方最新代码，router.js文件中的dynamicWrapper路径不支持sys/role写法。之前一直都没这个问题。
const routerConfig = {
    '/': {
      component: dynamicWrapper(app, ['user', 'login'], () => import('../layouts/BasicLayout')),
    },
    '/dashboard/analysis': {
      component: dynamicWrapper(app, ['chart'], () => import('../routes/Dashboard/Analysis')),
    },
    '/dashboard/monitor': {
      component: dynamicWrapper(app, ['monitor'], () => import('../routes/Dashboard/Monitor')),
    },
    '/dashboard/workplace': {
      component: dynamicWrapper(app, ['project', 'activities', 'chart'], () => import('../routes/Dashboard/Workplace')),
      // hideInBreadcrumb: true,
      // name: '工作台',
      // authority: 'admin',
    },
    '/form/basic-form': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/BasicForm')),
    },
    '/form/step-form': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm')),
    },
    '/form/step-form/info': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm/Step1')),
    },
    '/form/step-form/confirm': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm/Step2')),
    },
    '/form/step-form/result': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm/Step3')),
    },
    '/form/advanced-form': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/AdvancedForm')),
    },
    '/list/table-list': {
      component: dynamicWrapper(app, ['rule'], () => import('../routes/List/TableList')),
    },
    '/list/basic-list': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/BasicList')),
    },
    '/list/card-list': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/CardList')),
    },
    '/list/search': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/List')),
    },
    '/list/search/projects': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/Projects')),
    },
    '/list/search/applications': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/Applications')),
    },
    '/list/search/articles': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/Articles')),
    },
    '/profile/basic': {
      component: dynamicWrapper(app, ['profile'], () => import('../routes/Profile/BasicProfile')),
    },
    '/profile/advanced': {
      component: dynamicWrapper(app, ['profile'], () => import('../routes/Profile/AdvancedProfile')),
    },
    '/result/success': {
      component: dynamicWrapper(app, [], () => import('../routes/Result/Success')),
    },
    '/result/fail': {
      component: dynamicWrapper(app, [], () => import('../routes/Result/Error')),
    },
    '/exception/403': {
      component: dynamicWrapper(app, [], () => import('../routes/Exception/403')),
    },
    '/exception/404': {
      component: dynamicWrapper(app, [], () => import('../routes/Exception/404')),
    },
    '/exception/500': {
      component: dynamicWrapper(app, [], () => import('../routes/Exception/500')),
    },
    '/exception/trigger': {
      component: dynamicWrapper(app, ['error'], () => import('../routes/Exception/triggerException')),
    },
    '/user': {
      component: dynamicWrapper(app, [], () => import('../layouts/UserLayout')),
    },
    '/user/login': {
      component: dynamicWrapper(app, ['login'], () => import('../routes/User/Login')),
    },
    '/user/register': {
      component: dynamicWrapper(app, ['register'], () => import('../routes/User/Register')),
    },
    '/user/register-result': {
      component: dynamicWrapper(app, [], () => import('../routes/User/RegisterResult')),
    },
    # '/sys/user': {
      component: dynamicWrapper(app, ['sys/user'], () => import('../routes/Sys/User/index')),
    },
    '/sys/role': {
      component: dynamicWrapper(app, ['sys/role'], () => import('../routes/Sys/Role/index')),
    },
    '/sys/menu': {
      component: dynamicWrapper(app, ['sys/menu'], () => import('../routes/Sys/Menu/index')),
    },
    '/sys/office': {
      component: dynamicWrapper(app, ['sys/office'], () => import('../routes/Sys/Office/index')),
    },
    '/sys/session': {
      component: dynamicWrapper(app, ['sys/session'], () => import('../routes/Sys/Session/index')),
    },
    '/sys/log': {
      component: dynamicWrapper(app, ['sys/log'], () => import('../routes/Sys/Log/index')),
    },
  };

编译和启动都没错误，启动服务后，打开浏览器访问http://localhost:8000,页面提示：
×
[app.model] namespace should be unique
▶ 4 stack frames were collapsed.
(anonymous function)
D:/react_work/mms-design/src/common/router.js:25
  22 | _models.forEach(function (model) {
  23 |   if (modelNotExisted(app, model)) {
  24 |     // eslint-disable-next-line
> 25 |     app.model(require("../models/".concat(model)).default);
  26 |   }
  27 | });
  28 | 
View compiled
dynamicWrapper
D:/react_work/mms-design/src/common/router.js:22
  19 | // () => require('module')
  20 | // transformed by babel-plugin-dynamic-import-node-sync
  21 | if (_component.toString().indexOf('.then(') < 0) {
> 22 |   _models.forEach(function (model) {
  23 |     if (modelNotExisted(app, model)) {
  24 |       // eslint-disable-next-line
  25 |       app.model(require("../models/".concat(model)).default);
View compiled
getRouterData
D:/react_work/mms-design/src/common/router.js:237
  234 |   })
  235 | },
  236 | '/business/order': {
> 237 |   component: dynamicWrapper(app, ['sys/menu'], function () {
  238 |     return require('../routes/Sys/Menu/index');
  239 |   })
  240 | },
View compiled
RouterConfig
D:/react_work/mms-design/src/router.js:23
  20 | function RouterConfig(_ref) {
  21 |   var history = _ref.history,
  22 |       app = _ref.app;
> 23 |   var routerData = getRouterData(app);
  24 |   var UserLayout = routerData['/user'].component;
  25 |   var BasicLayout = routerData['/'].component;
  26 |   return React.createElement(_LocaleProvider, {
View compiled
▶ 22 stack frames were collapsed.
(anonymous function)
D:/react_work/mms-design/src/index.js:80
  77 | })(); // 5. Start
  78 | 
  79 | 
> 80 | app.start('#root');
  81 | FastClick.attach(document.body);
  82 | 
  83 |

嗯 ，这个是我router.js 文件内容
import { createElement } from 'react';
import dynamic from 'dva/dynamic';
import { getMenuData } from './menu';

let routerDataCache;

const modelNotExisted = (app, model) => (
  // eslint-disable-next-line
  !app._models.some(({ namespace }) => namespace === model)
);

// wrapper of dynamic
const dynamicWrapper = (app, models, component) => {
  // () => require('module')
  // transformed by babel-plugin-dynamic-import-node-sync
  if (component.toString().indexOf('.then(') < 0) {
    models.forEach((model) => {
      if (modelNotExisted(app, model)) {
        // eslint-disable-next-line
        app.model(require(`../models/${model}`).default);
      }
    });
    return (props) => {
      if (!routerDataCache) {
        routerDataCache = getRouterData(app);
      }
      return createElement(component().default, {
        ...props,
        routerData: routerDataCache,
      });
    };
  }
  // () => import('module')
  return dynamic({
    app,
    models: () => models.filter(
      model => modelNotExisted(app, model)).map(m => import(`../models/${m}.js`)
    ),
    // add routerData prop
    component: () => {
      if (!routerDataCache) {
        routerDataCache = getRouterData(app);
      }
      return component().then((raw) => {
        const Component = raw.default || raw;
        return props => createElement(Component, {
          ...props,
          routerData: routerDataCache,
        });
      });
    },
  });
};

function getFlatMenuData(menus) {
  let keys = {};
  menus.forEach((item) => {
    if (item.children) {
      keys[item.path] = { ...item };
      keys = { ...keys, ...getFlatMenuData(item.children) };
    } else {
      keys[item.path] = { ...item };
    }
  });
  return keys;
}

export const getRouterData = (app) => {
  const routerConfig = {
    '/': {
      component: dynamicWrapper(app, ['user', 'login'], () => import('../layouts/BasicLayout')),
    },
    '/dashboard/analysis': {
      component: dynamicWrapper(app, ['chart'], () => import('../routes/Dashboard/Analysis')),
    },
    '/dashboard/monitor': {
      component: dynamicWrapper(app, ['monitor'], () => import('../routes/Dashboard/Monitor')),
    },
    '/dashboard/workplace': {
      component: dynamicWrapper(app, ['project', 'activities', 'chart'], () => import('../routes/Dashboard/Workplace')),
      // hideInBreadcrumb: true,
      // name: '工作台',
      // authority: 'admin',
    },
    '/form/basic-form': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/BasicForm')),
    },
    '/form/step-form': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm')),
    },
    '/form/step-form/info': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm/Step1')),
    },
    '/form/step-form/confirm': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm/Step2')),
    },
    '/form/step-form/result': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/StepForm/Step3')),
    },
    '/form/advanced-form': {
      component: dynamicWrapper(app, ['form'], () => import('../routes/Forms/AdvancedForm')),
    },
    '/list/table-list': {
      component: dynamicWrapper(app, ['rule'], () => import('../routes/List/TableList')),
    },
    '/list/basic-list': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/BasicList')),
    },
    '/list/card-list': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/CardList')),
    },
    '/list/search': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/List')),
    },
    '/list/search/projects': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/Projects')),
    },
    '/list/search/applications': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/Applications')),
    },
    '/list/search/articles': {
      component: dynamicWrapper(app, ['list'], () => import('../routes/List/Articles')),
    },
    '/profile/basic': {
      component: dynamicWrapper(app, ['profile'], () => import('../routes/Profile/BasicProfile')),
    },
    '/profile/advanced': {
      component: dynamicWrapper(app, ['profile'], () => import('../routes/Profile/AdvancedProfile')),
    },
    '/result/success': {
      component: dynamicWrapper(app, [], () => import('../routes/Result/Success')),
    },
    '/result/fail': {
      component: dynamicWrapper(app, [], () => import('../routes/Result/Error')),
    },
    '/exception/403': {
      component: dynamicWrapper(app, [], () => import('../routes/Exception/403')),
    },
    '/exception/404': {
      component: dynamicWrapper(app, [], () => import('../routes/Exception/404')),
    },
    '/exception/500': {
      component: dynamicWrapper(app, [], () => import('../routes/Exception/500')),
    },
    '/exception/trigger': {
      component: dynamicWrapper(app, ['error'], () => import('../routes/Exception/triggerException')),
    },
    '/user': {
      component: dynamicWrapper(app, [], () => import('../layouts/UserLayout')),
    },
    '/user/login': {
      component: dynamicWrapper(app, ['login'], () => import('../routes/User/Login')),
    },
    '/user/register': {
      component: dynamicWrapper(app, ['register'], () => import('../routes/User/Register')),
    },
    '/user/register-result': {
      component: dynamicWrapper(app, [], () => import('../routes/User/RegisterResult')),
    },
    '/sys/log': {
      component: dynamicWrapper(app, ['sys'], () => import('../routes/Sys/Log/index')),
    },
    // '/user/:id': {
    //   component: dynamicWrapper(app, [], () => import('../routes/User/SomeComponent')),
    // },
  };
  // Get name from ./menu.js or just set it in the router data.
  const menuData = getFlatMenuData(getMenuData());
  const routerData = {};
  Object.keys(routerConfig).forEach((item) => {
    const menuItem = menuData[item.replace(/^\//, '')] || {};
    routerData[item] = {
      ...routerConfig[item],
      name: routerConfig[item].name || menuItem.name,
      authority: routerConfig[item].authority || menuItem.authority,
    };
  });
  return routerData;
};
========================================================
启动后浏览器提示
Error: Cannot find module './sys'.
▶ 2 stack frames were collapsed.
(anonymous function)
D:/react_work/mms-design/src/common/router.js:25
  22 | _models.forEach(function (model) {
  23 |   if (modelNotExisted(app, model)) {
  24 |     // eslint-disable-next-line
> 25 |     app.model(require("../models/".concat(model)).default);
  26 |   }
  27 | });
  28 | 
View compiled
dynamicWrapper
D:/react_work/mms-design/src/common/router.js:22
  19 | // () => require('module')
  20 | // transformed by babel-plugin-dynamic-import-node-sync
  21 | if (_component.toString().indexOf('.then(') < 0) {
> 22 |   _models.forEach(function (model) {
  23 |     if (modelNotExisted(app, model)) {
  24 |       // eslint-disable-next-line
  25 |       app.model(require("../models/".concat(model)).default);
View compiled
getRouterData
D:/react_work/mms-design/src/common/router.js:232
  229 |   })
  230 | },
  231 | '/sys/log': {
> 232 |   component: dynamicWrapper(app, ['sys'], function () {
  233 |     return require('../routes/Sys/Log/index');
  234 |   })
  235 | } // '/user/:id': {
View compiled
RouterConfig
D:/react_work/mms-design/src/router.js:23
  20 | function RouterConfig(_ref) {
  21 |   var history = _ref.history,
  22 |       app = _ref.app;
> 23 |   var routerData = getRouterData(app);
  24 |   var UserLayout = routerData['/user'].component;
  25 |   var BasicLayout = routerData['/'].component;
  26 |   return React.createElement(_LocaleProvider, {
View compiled
▶ 22 stack frames were collapsed.
(anonymous function)
D:/react_work/mms-design/src/index.js:80
  77 | })(); // 5. Start
  78 | 
  79 | 
> 80 | app.start('#root');
  81 | FastClick.attach(document.body);
  82 | 
  83 | 
View compiled
▶ 6 stack frames were collapsed.

--- a/src/common/router.js
+++ b/src/common/router.js
@@ -6,7 +6,9 @@ let routerDataCache;
 
 const modelNotExisted = (app, model) => (
   // eslint-disable-next-line
-  !app._models.some(({ namespace }) => namespace === model)
+  !app._models.some(({ namespace }) => {
+    return namespace === model.substring(model.lastIndexOf('/') + 1);
+  })
 );
 
 // wrapper of dynamic

